src/Config/Converters/McpOptionsConverterFactory.cs

```diff
@@ -0,0 +1,169 @@
+                        case "describe-entities":
+                            describeEntities = reader.GetBoolean();
+                            break;
+                        case "create-record":
+                            createRecord = reader.GetBoolean();
+                            break;
+                        case "read-record":
+                            readRecord = reader.GetBoolean();
+                            break;
+                        case "update-record":
+                            updateRecord = reader.GetBoolean();
+                            break;
+                        case "delete-record":
+                            deleteRecord = reader.GetBoolean();
+                            break;
+                        case "execute-record":
+                            executeRecord = reader.GetBoolean();
+                            break;
+                        default:
+                            throw new JsonException($"Unexpected property '{property}' in dml-tools configuration.");
+                    }
+                }
+            }
+
+            throw new JsonException("DML Tools configuration is missing closing brace.");
+        }
+
+        /// <summary>
+        /// Writes DmlToolsConfig to JSON.
+        /// If all tools have the same state, writes as boolean.
+        /// Otherwise, writes as an object with individual tool settings.
+        /// </summary>
+        public override void Write(Utf8JsonWriter writer, DmlToolsConfig? value, JsonSerializerOptions options)
+        {
+            if (value is null)
+            {
+                writer.WriteNullValue();
+                return;
+            }
+
+            // Check if this can be simplified to a boolean
+            bool hasIndividualSettings = value.DescribeEntities.HasValue ||
+                                       value.CreateRecord.HasValue ||
+                                       value.ReadRecord.HasValue ||
+                                       value.UpdateRecord.HasValue ||
+                                       value.DeleteRecord.HasValue ||
+                                       value.ExecuteRecord.HasValue;
+
+            if (!hasIndividualSettings)
+            {
+                writer.WriteBooleanValue(value.AllToolsEnabled);
+            }
+            else
+            {
+                writer.WriteStartObject();
+
+                if (value.DescribeEntities.HasValue)
+                {
+                    writer.WriteBoolean("describe-entities", value.DescribeEntities.Value);
+                }
+
+                if (value.CreateRecord.HasValue)
+                {
+                    writer.WriteBoolean("create-record", value.CreateRecord.Value);
+                }
+
+                if (value.ReadRecord.HasValue)
+                {
+                    writer.WriteBoolean("read-record", value.ReadRecord.Value);
+                }
+
+                if (value.UpdateRecord.HasValue)
```

Comment:
We should use the corresponding flags: `UserProvidedUpdateRecord` to determine whether to write the property to the file or not while serializing. Since we are setting the `UpdateRecord` to a default value always, we will end up writing the property always even if it was not provided by the user.